activesupport/lib/active_support/cache.rb

```diff
@@ -85,6 +85,14 @@ def expand_cache_key(key, namespace = nil)
         expanded_cache_key
       end
 
+      def expand_cache_version(key)
+        case
+        when key.respond_to?(:cache_version) then key.cache_version
```

Comment:
We need to have a consistent behavior for the cache version. `cache_key` can only be a string at the end (meaning that String is always put into cache store). Maybe we need the same for version. In this case we always need to call `to_param` here.